core/src/nodes/protocols_handler.rs

```diff
@@ -92,15 +105,22 @@ pub trait ProtocolsHandler {
     /// >           context you wouldn't accept one in particular (eg. only allow one substream at
     /// >           a time for a given protocol). The reason is that remotes are allowed to put the
     /// >           list of supported protocols in a cache in order to avoid spurious queries.
-    fn listen_protocol(&self) -> Self::Protocol;
+    fn listen_protocol(&self) -> Self::InboundProtocol;
+
+    fn dialer_protocol(&self) -> Self::OutboundProtocol;
```

Comment:
This method is useless, as the protocol is created when requesting a new substream.

Reply:
It is used here: https://github.com/libp2p/rust-libp2p/pull/612/files#diff-ac7fb68c0f3a5c9549b6b90f3732dc00R934 as `InboundProtocol` and `OutboundProtocol` are different types.

Reply:
I see. This is a bad design from `Toggleable`. It should be possible to create an equivalent of `None` without actually building a protocol.
I really want this PR to be merged soon, so it could be in a further PR.